src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,102 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double ARRAY_MULTIPLIER = 1.5;
+    private T[] elementData;
+    private int arraySize;
+
+    public ArrayList() {
+        elementData = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        growIfArrayFull();
+        elementData[arraySize] = value;
+        arraySize++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        growIfArrayFull();
+        if (index < arraySize && index >= 0 || index == arraySize) {
+            System.arraycopy(elementData, index, elementData, index + 1, arraySize - index);
+            elementData[index] = value;
+            arraySize++;
+            return;
+        }
+        throw new ArrayListIndexOutOfBoundsException("Index passed to method is invalid!");
```

Comment:
```suggestion
        throw new ArrayListIndexOutOfBoundsException("Index " + index + " is invalid!");
```
![image](https://github.com/mate-academy/jv-array-list/assets/20326858/338eb4ba-d48e-405c-8d8e-a4a4f3d46d0d)